tango/common/logging.py

```diff
@@ -114,3 +116,78 @@ def initialize_logging(
         FILE_FRIENDLY_LOGGING = True
         os.environ["FILE_FRIENDLY_LOGGING"] = "true"
         click_logger.disabled = True
+
+
+def logging_tqdm(
```

Comment:
1. You don't have to set an environment variable. It's always on.
2. It does not redirect `stderr` in a dodgy way.
3. Log messages get written to your logger, not the global `"tqdm"` logger.
3. It does not depend on implementation details of the original tqdm.
4. It is less code.

Reply:
Let me rephrase. What I'm really asking is this: (A) why not implement this using our existing `Tqdm` wrapper, and (B) why have two separate approaches for file friendly progress bars?

For (A), your 2nd point is still valid, but your approach of implementing a new tqdm from scratch adds more code that looks fairly tricky. We should at least have some test coverage here.

But to emphasize (B) again, it seems like there is a lot of overlap in use-cases here for you `logging_tqdm` and the existing `Tqdm` wrapper with `FILE_FRIENDLY_LOGGING` on. I'd rather go with one or the other. Maybe our file-friendly version of the `Tqdm` wrapper uses your `logging_tqdm` code instead of `tqdm` internals.

Reply:
I'm happy to write tests for it, but only if we decide to actually use it.

I looked at our current usage of TQDM, and the only feature that `logging_tqdm` is missing is `wrapattr()`. There are a few others, but those only apply to rendering visual progress bars, so we don't have to care about them.

Overall I think `logging_tqdm()` is the superior solution, but there is a lot to do, and the success of Tango won't hinge on the quality of the progress bars. So I'll take this thing out and replace it with `FILE_FRIENDLY_LOGGING`.

Reply:
c3a743b